executor/src/core.rs

```diff
@@ -106,35 +104,41 @@ where
 
     /// Execute a single certificate.
     async fn execute_certificate(&mut self, message: &ConsensusOutput) -> SubscriberResult<()> {
-        // Skip the certificate if it contains no transactions.
-        if message.certificate.header.payload.is_empty() {
-            self.execution_indices.skip_certificate();
-            return Ok(());
+        // Collect all transactions in all the batches.
+        let mut batches = Vec::new();
+
+        for batch_digest in message.certificate.header.payload.keys() {
+            batches.push(self.collect_batch(batch_digest).await?);
```

Comment:
In this instance it would mean stopping the `Core::run` altogether with the `"Failed to run core"` message. 

I know it's not what you want, thanks for bringing it up. See my [other comment](https://github.com/MystenLabs/narwhal/pull/818/files/6432daae2e69d813942837c058d34e0865192035#r949976679) about why it's difficult to say what the right thing to do is. 

For example in Tendermint you have some guarantees about the content because transactions have to conform to Protobuf message formats to be admitted into the blocks, so this problem doesn't come up. 

Arguably if you _know_ what type of `Transaction` you are dealing with, ie. you aren't passing raw bytes to `ExecutionState`, then you might want to think about validating their content up front before batching, and also after downloading it, before casting a vote on the batch. That way if it fails here, we are right to stop because it must in a minority who can't deal with something everyone else okayed already.

But if you trust that most of your nodes will run the same version of the software and you aren't doing any vetting of input from clients, then you're right it's best to skip rubbish transactions than to halt all validators.

Reply:
Changed the transaction deserialization to skipping. Now we only stop if a batch cannot be retrieved from the store.